### PR TITLE
Initial conversion from Rnw to Rmd

### PR DESCRIPTION
**Pull request by villafup**
@Bioconductor/sweave2rmd The original has a line break in the **_package_** section of the yaml.  I couldn't get it to work, so I added it as a sidenote.  Do any of you know how to put a line break here?  Wasn't able to google anything useful. 

[Original Vignette](https://github.com/Bioconductor/GSEABase/files/9499760/GSEABase.pdf)

### **_Original_**
Original

### **_Converted_**
Converted

### CODE REVIEW

**Comment by mtmorgan** [comment]
I think these commits should remove the original Rnw file (else R CMD build will not necessarily do what is expected) and should add the translator to the DESCRIPTION file as a contributor with comment about translating vignettes from Rnw to Rmd. Perhaps this would involve changes to the `Authors@R` format.

The current translation looks great!

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Fix "role" on line 8.

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: This line should be removed and we should add to the `Authors@R` vector a `person` entry for the `Bioconductor Package Maintainer` as in https://github.com/Bioconductor/graph/blob/7522c5c9bf227067bd02bc76f7ef7668b77baa68/DESCRIPTION#L12-L15. This is a consequence of using the `Authors@R`; otherwise, we'll get an error when building the package.

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: The version should be updated to the current devel version, 1.61.0. This was likely created before the release, but we'll always merge with devel (it's less work). I'll deal with the conflict this generates with the `DESCRIPTION` file before merging.

I think the footnote doesn't work here because it's the YAML part of the document. We can do the following that will look similar to the PDF:

```
package: |
  GSEABase 1.59.0

  Report issues on https://github.com/Bioconductor/GSEABase/issues
```

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: `yes` should be `true`. I made an issue in sweave2rmd to correct this misinfo.

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: Remove space between `GeneSet` and period.

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: Missing the name "AnnotationIdentifier" before the period.

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: Need space after `verbose`

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: Insert `hgu95av2` after "chip".

File: vignettes/GSEABase.Rmd
Comment (jwokaty) [inline vignettes/GSEABase.Rmd]: This section lost a few words. It should read

```
Remember that `sample.ExpressionSet` contains just 500 probe sets, so the small size of the
subset is not surprising. Note also that subsetting required mapping of symbol identifiers in
bgs to AnnotationIdentifiers; this map used the annotation information in the expression set,
and is not necessarily 1:1.
```

**Review by jwokaty — CHANGES_REQUESTED** [review]
In addition to the inline requested, the .Rnw should be removed as Martin suggested then I believe we can resolve the conflict and merge.

**Comment by jwokaty** [comment]
- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [x] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [x] any added lines use the same spacing and indents as the existing
  document
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

**Comment by mtmorgan** [comment]
FWIW, as a further sanity check I created the R script that is implied by the Rnw vignette (before applying the PR)
```
R CMD Stangle vignettes/GSEABase.Rnw
mv GSEABase.R GSEABase-Rnw.R
```
and after the PR
```
R CMD Stangle vignettes/GSEABase.Rmd
mv GSEABase.R GSEABase-Rmd.R
```
Then I compared the two files, ignoring spaces
```
$ diff -w GSEABase-Rmd.R GSEABase-Rnw.R
0a1,4
> ## ----style-knitr, eval=TRUE, echo=FALSE, results="asis"-----------------------
> BiocStyle::latex()
>
>
62d65
<
124d126
<
151c153
< ## ----ReportingTools, echo=TRUE------------------------------------------------
---
> ## ----ReportingTools-----------------------------------------------------------
$
```
Manually inspecting this, I see that there are NO MEANINGFUL CHANGES in the evaluated R code.

This is great, thanks so much for this translation!

**Comment by jwokaty** [comment]
> FWIW, as a further sanity check I created the R script that is implied by the Rnw vignette (before applying the PR)
> 
> ```
> R CMD Stangle vignettes/GSEABase.Rnw
> mv GSEABase.R GSEABase-Rnw.R
> ```
> 
> and after the PR
> 
> ```
> R CMD Stangle vignettes/GSEABase.Rmd
> mv GSEABase.R GSEABase-Rmd.R
> ```
> 
> Then I compared the two files, ignoring spaces
> 
> ```
> $ diff -w GSEABase-Rmd.R GSEABase-Rnw.R
> 0a1,4
> > ## ----style-knitr, eval=TRUE, echo=FALSE, results="asis"-----------------------
> > BiocStyle::latex()
> >
> >
> 62d65
> <
> 124d126
> <
> 151c153
> < ## ----ReportingTools, echo=TRUE------------------------------------------------
> ---
> > ## ----ReportingTools-----------------------------------------------------------
> $
> ```
> 
> Manually inspecting this, I see that there are NO MEANINGFUL CHANGES in the evaluated R code.
> 
> This is great, thanks so much for this translation!

@mtmorgan Thanks for sharing this. I've been thinking a lot about how we can automate and improve the process since it's very manual and requires careful attention to detail. I never used `Stangle` and I think this is a great tool for sweave2rmd.

**Pull request merged**
